api/service/metrics/storage.go

```diff
@@ -0,0 +1,121 @@
+package metrics
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"sync"
+
+	"github.com/ethereum/go-ethereum/ethdb"
+	"github.com/ethereum/go-ethereum/rlp"
+
+	"github.com/harmony-one/harmony/internal/ctxerror"
+	"github.com/harmony-one/harmony/internal/utils"
+)
+
+// Constants for storage.
+const (
+	ConnectionsNumberPrefix     = "cnp"
+	ConsensusFramePrefix        = "cfp"
+	CurrentConnectionsNumberKey = "cnk"
+	BalancePrefix               = "bp"
+	BlocksProcessedPrefix       = "bpp"
+	BlocksSuccessPrefix         = "bsp"
+	LeaderTimePrefix            = "ltp"
+	NodeCPUPrefix               = "ncp"
+	NodeTrafficPrefix           = "ntp"
+	TranscationsProcessedPrefix = "tpp"
+	TransactionsSuccessPrefix   = "tsp"
+)
+
+// GetConnectionsNumberKey ...
+func GetConnectionsNumberKey(moment int) string {
```

Comment:
is there a better design pattern of the API that we can use, instead of adding functions one by one for each metrics?

Reply:
Consider using [Go Interfaces](https://gobyexample.com/interfaces)

Reply:
Sure is, we can variadic params list, which will define metric name, params types, values, then parse it. I don't like this approach, because it adds additional layer of complexity.

Reply:
go interfaces is actually a good design pattern here.  It is another layer, but it is more extensible.